docs/en/sql-reference/functions/string-search-functions.md

```diff
@@ -852,16 +852,42 @@ multiFuzzyMatchAllIndices(haystack, distance, \[pattern<sub>1</sub>, pattern<sub
 
 ## extract
 
-Extracts a fragment of a string using a regular expression. If `haystack` does not match the `pattern` regex, an empty string is returned.
-
-For regex without subpatterns, the function uses the fragment that matches the entire regex. Otherwise, it uses the fragment that matches the first subpattern.
+Extracts a fragment of a string using a regular expression.
+If `haystack` does not match the `pattern` regex, an empty string is returned. 
+For a regex without sub-patterns, the function uses the fragment that matches the entire regex. 
+Otherwise, it uses the fragment that matches the first subpattern.
 
 **Syntax**
 
 ```sql
 extract(haystack, pattern)
 ```
 
+*Arguments**
+
+- `haystack` — Input string. [String](../data-types/string.md).
+- `pattern` — Regular expression with [re2 syntax](https://github.com/google/re2/wiki/Syntax).
+
+**Returned value**
+
+- The fragment of a string matching the provided pattern. [String](../data-types/string.md).
+
+**Example**
+
+Query:
+
+```sql
+SELECT extract('number: 1, number: 2, number: 3', '\\d+') AS result;
```

Comment:
It would be nice to use the same haystack / pattern example here and in the docs of function `extractAll`